pxr/usdImaging/usdImaging/primAdapter.cpp

```diff
@@ -863,6 +862,7 @@ _GetDirtyBitsForPrimvarChange(
             dirty = HdChangeTracker::DirtyPrimvar;
```

Comment:
Instead of moving the `PrimvarChangeAdd` case I did try to use `HdChangeTracker::DirtyComputationPrimvarDesc`. I wasn't sure if this was the bit that the comment might have been eluding to. However, that didn't seem to fix the issue for the `widths` attribute

Reply:
Your PR made me examine the current code, and I believe we should return the dirty bit passed in (`valueChangeDirtyBit`) for all 4 scenarios (primvar added, primvar removed, primvar value change, primvar descriptor change).
We test against each or all of the primvar dirty bits in `HdChangeTracker::IsPrimvarDirty` and `HdChangeTracker::IsAnyPrimvarDirty` respectively. So, if we were to delete an authored widths primvar, instead of `DirtyPrimvar`, we ought to return `DirtyWidths`.

We can simplify and correct the logic in `primAdapter.cpp` as follows:
- Delete `_GetDirtyBitsForPrimvarChange`
- Remove `PrimvarChange` enum and return `void` in `_ProcessPrimvarChange` or fold this logic in the caller sites.
- Rename `valueChangeDirtyBit` to `primvarDirtyBit` and simply return it in `_ProcessNonPrefixedPrimvarPropertyChange` and `_ProcessPrefixedPrimvarPropertyChange`.

If that makes sense, could you please make these changes to the PR? Thanks!

Reply:
Thank you for the notes and checking out the PR! Updated according to notes